frontend/gateway/gateway.go

```diff
@@ -1383,10 +1374,28 @@ func (lbf *llbBridgeForwarder) convertRef(id string) (solver.ResultProxy, error)
 	if !ok {
 		return nil, errors.Errorf("return reference %s not found", id)
 	}
-
 	return r, nil
 }
 
+func (lbf *llbBridgeForwarder) convertRefShared(id string) (solver.ResultProxy, error) {
```

Comment:
I think this can be called `cloneRef` (Shared only makes sense for a type that has a reference counter on it).

Just to clean this up, there seems to be only one place that calls `convertRef` that could be also made to use this with extra release call. (In another PR seems that container releasing needs some fixes as well).

Reply:
Sounds good, will do a follow-up to this to tidy up around the container releasing, I'm definitely less familiar with it.